### ISSUE TITLE
Lighthouse: Accessibility - Forms

### ISSUE THREAD

**Issue by ExperimentsInHonesty** [hackforla/tdm-calculator#1619]
### Overview
In order for your sites form(s) to be usable by visitors using screen readers all the form <input> elements need labels.  There are specific details and exceptions, which can be found in the instructions below.

### Action Items 
- [ ] If your site already has forms, review the instructions and document the changes needed to bring - - [ ] If your site does not have forms, review the instructions and design new forms using the WCAG standards. 

### Instructions
Deque University 
https://dequeuniversity.com/rules/axe/3.2/label

**Comment by ExperimentsInHonesty** [comment]
We will have to figure out another way to test the forms accessibility, since the lighthouse chrome extension just calls the page url, and, given that can't consent to the Privacy Policy, it just checks that every time.

**Comment by DrAcula27** [comment]
ETA: 14 April
Availability: weekdays after 5pm Pacific time

**Comment by ryanfchase** [comment]
@DrAcula27 asked a question during the 2025-04-09 TDM dev meeting:
- Q: is there is a preferred way to document the issues reported from WAVE?
  - A: Can we see an example of a generated WAVE report for one of the forms? This will solidify our approach to documenting our findings from the WAVE extnesion.
    - Short term let's have you add a comment that details all of the problems addressed by each WAVE report. We _could_ do a Google Sheet too. 
    - Long term, perhaps we could create github issue templates that allow us to hand off these WAVE reports to designers.

**Comment by DrAcula27** [comment]
@ryanfchase 

<details><summary>Example of generated WAVE report</summary>
![Image](https://github.com/user-attachments/assets/e28fb8f8-8090-4b1b-a9ae-69f1ea073c4b)
</details>

I was thinking to add these screenshots from each page as a detail dropdown, either in the issue itself or as a comment (or maybe one comment per page?) on the issue.

**Comment by ryanfchase** [comment]
@DrAcula27 ok, yeah do one of these screenshots for each page. I think you should do one comment per page. Then later, as a batch processing step, we can figure out what information we want to extract from each of those comments. We can move them to new issues, or compile into a spreadsheet... but for now just do one comment per screenshot.

**Comment by DrAcula27** [comment]
#### [WAVE Results for Feedback Page](https://tdm-dev.azurewebsites.net/feedback)

![Image](https://github.com/user-attachments/assets/d1574247-b489-4357-8470-6a79f1293f4a)

- Errors
  - 1 X Empty button
- Contrast Errors
  - 2 X Very low contrast
- Alerts
  - 2 X Noscript element
  - 1 X Underlined text
- Features
   - 1 X Linked image with alt text
   - 4 X Form label
   - 1 X Language
- Structural Elements
  - 1 X Heading level 1
  - 1 X Unordered list
  - 1 X Inline frame
  - 1 X Footer
- ARIA
  - 1 X ARIA
  - 4 X ARIA hidden

**Comment by DrAcula27** [comment]
#### [WAVE Results for Create Project, Page 1](https://tdm-dev.azurewebsites.net/calculation/1/0)

![Image](https://github.com/user-attachments/assets/86f4d129-a958-492d-bad6-6e21206aa717)

- Errors
  - 1 X Missing form label
  - 3 X Empty button
  - 1 X Broken ARIA reference
- Alerts
  - 1 X Orphaned form label
  - 1 X Skipped heading level
  - 1 X Possible heading
  - 2 X Noscript element
  - 1 X Underlined text
  - 1 X Layout table
- Features
  - 1 X Alternative text
  - 1 X Linked image with alternative text
  - 7 X Form label
  - 1 X Language
- Structural Elements
  - 1 X Heading level 1
  - 4 X Heading level 3
  - 1 X Unordered list
  - 1 X Inline frame
  - 1 X Footer
- ARIA
  - 2 X ARIA
  - 1 X ARIA alert or live region
  - 4 X ARIA hidden

**Comment by DrAcula27** [comment]
[WAVE Results for Create Project, Page 2](https://tdm-dev.azurewebsites.net/calculation/2/0)

![Image](https://github.com/user-attachments/assets/1ff67c24-d17a-4e11-839c-fe62a36274eb)

- Errors
  - 3 X Empty button
  - 2 X Broken ARIA reference
- Alerts
  - 1 X Skipped heading level
  - 1 X Possible heading
  - 2 X Noscript element
  - 1 X Underlined text
  - 1 X Layout table
- Features
  - 1 X Alternative text
  - 1 X Linked image with alternative text
  - 24 X Form label
  - 1 X Language
- Structural Elements
  - 1 X Heading level 1
  - 4 X Heading level 3
  - 8 X Heading level 4
  - 1 X Unordered list
  - 1 X Inline frame
  - 1 X Footer
- ARIA
  - 4 X ARIA hidden

**Comment by DrAcula27** [comment]
#### [WAVE Results for Create Project, Page 3](https://tdm-dev.azurewebsites.net/calculation/3/0)

![Image](https://github.com/user-attachments/assets/2a757c72-da45-4ccb-a82c-c2fd86aada11)

- Errors
  - 3 X Empty button
  - 4 X Broken ARIA reference
- Alerts
  - 1 X Skipped heading level
  - 1 X Possible heading
  - 2 X Noscript element
  - 1 X Underlined text
  - 1 X Layout table
- Features
  - 1 X Alternative text
  - 1 X Linked image with alternative text
  - 1 X Form label
  - 1 X Language
- Structural Elements
  - 1 X Heading level 1
  - 3 X Heading level 3
  - 2 X Heading level 4
  - 1 X Unordered list
  - 1 X Inline frame
  - 1 X Footer
- ARIA
  - 4 X ARIA hidden

**Comment by DrAcula27** [comment]
#### [WAVE Results for Create Project, Page 4](https://tdm-dev.azurewebsites.net/calculation/4/0)

![Image](https://github.com/user-attachments/assets/1a31d083-837c-44ad-bf1b-3b8924807010)

- Errors
  - 1 X Missing form label
  - 3 X Empty button
  - 2 X Broken ARIA reference
- Alerts
  - 2 X Orphaned form label
  - 2 X Skipped heading level
  - 1 X Possible heading
  - 2 X Noscript element
  - 1 X Underlined text
  - 1 X Layout table
- Features
  - 1 X Alternative text
  - 1 X Linked image with alternative text
  - 36 X Form label
  - 1 X Language
- Structural Elements
  - 2 X Heading level 1
  - 3 X Heading level 3
  - 14 X Heading level 4
  - 1 X Unordered list
  - 1 X Inline frame
  - 1 X Footer
- ARIA
  - 33 X ARIA
  - 15 X ARIA tabindex
  - 15 X ARIA alert or live region
  - 34 X ARIA hidden
  - 15 X ARIA expanded
  - 15 X ARIA popup

**Comment by ryanfchase** [comment]
Thanks @DrAcula27! Can you update the ticket with the following?
- Review the original post and check off the boxes that you believe are done
- If all the action items are complete, move ticket to Technical Review and add `ready for DEV Lead - 311` ?

I mentioned that we could do batch processing on each of these comments. I will want to review your work with the TDM leads to decide what to do next. Thanks!

**Comment by DrAcula27** [comment]
@ryanfchase I still have a few pages to do. I will get to that tonight, but basically, the last 3 checkboxes need to be done for the following pages:

Auth:
- [Login](https://tdm-dev.azurewebsites.net/login)
- [Register](https://tdm-dev.azurewebsites.net/register)
- [Forgot Password](https://tdm-dev.azurewebsites.net/forgotpassword)

Pages only available to logged-in user:
- [Update Your Account](https://tdm-dev.azurewebsites.net/updateaccount/%3Cuser@email.com%3E)
- [My Projects](https://tdm-dev.azurewebsites.net/projects)
- [Submit Snapshot](https://tdm-dev.azurewebsites.net/submit)

**Comment by DrAcula27** [comment]
#### [WAVE Results for Login Page](https://tdm-dev.azurewebsites.net/login)

![Image](https://github.com/user-attachments/assets/04b471a9-20ed-4b82-a4e4-33f7e5e82aa8)

Errors
- 2 X Missing form label
- 1 X Empty button

Alerts
- 1 X No heading structure
- 2 X Noscript element
- 1 X Underlined text

Features
- 1 X Linked image with alternative text
- 1 X Language

Structural Elements
- 1 X Unordered list
- 1 X Inline frame
- 1 X Footer

ARIA
- 4 X ARIA hidden

**Comment by DrAcula27** [comment]
#### [WAVE Results for Register Page](https://tdm-dev.azurewebsites.net/register)

![Image](https://github.com/user-attachments/assets/1bb30b89-eeba-4acd-92a0-3bc647dad46d)

Errors
- 5 X Missing form label
- 1 X Empty button

Contrast Errors
- 1 X Very low contrast

Alerts
- 1 X Skipped heading level
- 1 X Redundant link
- 2 X Noscript element
- 1 X Underlined text

Features
- 1 X Linked image with alternative text
- 1 X Language

Structural Elements
- 1 X Heading level 1
- 1 X Heading level 3
- 1 X Unordered list
- 1 X Inline frame
- 1 X Footer

ARIA
- 4 X ARIA hidden

**Comment by DrAcula27** [comment]
#### [WAVE Results for Forgot Password Page](https://tdm-dev.azurewebsites.net/forgotpassword)

![Image](https://github.com/user-attachments/assets/5acf2786-c836-44e4-a3d3-904c3b8b2f09)

Errors
- 1 X Missing form label
- 1 X Empty button

Contrast Errors
- 1 X Very low contrast

Alerts
- 1 X Skipped heading level
- 2 X Noscript element
- 1 X Underlined text

Features
- 1 X Linked image with alternative text
- 1 X Language

Structural Elements
- 1 X Heading level 1
- 2 X Heading level 3
- 1 X Unordered list
- 1 X Inline frame
- 1 X Footer

ARIA
- 4 X ARIA hidden

**Comment by DrAcula27** [comment]
#### [WAVE Results for Update Account Page](https://tdm-dev.azurewebsites.net/updateaccount/<user@email.com>)

![Image](https://github.com/user-attachments/assets/9aa46839-5fd2-4855-a56f-a3d2da0b71fc)

Errors
- 3 X Missing form label
- 1 X Empty button

Alerts
- 2 X Noscript element
- 1 X Underlined text

Features
- 1 X Linked image with alternative text
- 1 X Language

Structural Elements
- 1 X Heading level 1
- 1 X Unordered list
- 1 X Inline frame
- 1 X Footer

ARIA
- 4 X ARIA hidden

**Comment by DrAcula27** [comment]
#### [WAVE Results for My Projects Page](https://tdm-dev.azurewebsites.net/projects)

![Image](https://github.com/user-attachments/assets/99d25edb-001b-498b-8a29-9618a475b7fe)

Errors
- 4 X Missing form label
- 7 X Empty button
- 2 X Empty table header
- 1 X Broken ARIA reference

Alerts
- 1 X Skipped heading level
- 2 X Noscript element
- 1 X Underlined text
- 1 X Layout table

Features
- 1 X Alternative text
- 1 X Linked image with alternative text
- 1 X Language

Structural Elements
- 1 X Data table
- 10 X Table header cell
- 2 X Heading level 1
- 2 X Heading level 3
- 3 X Unordered list
- 1 X Inline frame
- 1 X Footer

ARIA
- 3 X ARIA
- 1 X ARIA description
- 1 X ARIA tabindex
- 1 X ARIA alert or live region
- 6 X ARIA hidden
- 1 X ARIA expanded
- 1 X ARIA popup

**Comment by DrAcula27** [comment]
#### [WAVE Results for Submit Snapshot Page](https://tdm-dev.azurewebsites.net/submit)

![Image](https://github.com/user-attachments/assets/887e7e2c-b225-4dae-b9c9-1f1e520635de)

Errors
- 1 X Empty button
- 1 X Empty table header

Alerts
- 1 X Skipped heading level
- 2 X Noscript element
- 1 X Underlined text

Features
- 1 X Linked image with alternative text
- 1 X Language

Structural Elements
- 3 X Data table
- 12 X Table header cell
- 1 X Heading level 1
- 3 X Heading level 3
- 1 X Unordered list
- 1 X Inline frame
- 1 X Footer

ARIA
- 1 X ARIA
- 4 X ARIA hidden

**Comment by ryanfchase** [comment]
ETA for review: 2025-04-16 6pm PT (tonight)

**Comment by DrAcula27** [comment]
@entrotech, per discussion with @ryanfchase, we’d like to go ahead and close this ticket so I can receive credit for the work completed, and then open a new Epic to continue this effort with sub-issues.

**Comment by ExperimentsInHonesty** [comment]
@DrAcula27 I agree with making an epic and subissues.  Are you doing that, or someone else?

**Comment by ExperimentsInHonesty** [comment]
@ryanfchase If Danielle is going to make the epic and sub issues, this issue can be converted into the epic, and then action items updated.  If You or some other developer is going to make the epic and the sub issues, then Get the epic started and link this issue to it as background and close this issue.

**Comment by ryanfchase** [comment]
I will be making the epic. I've gotten started and provided an ETA for completion. Linking this ticket to the epic for reference and closing this ticket as complete.
- https://github.com/hackforla/tdm-calculator/issues/2410
